e2e/testcases/admission_test.go

```diff
@@ -141,55 +147,96 @@ metadata:
 }
 
 func TestDisableWebhookConfigurationUpdate(t *testing.T) {
-	webhook := []byte(`
-apiVersion: admissionregistration.k8s.io/v1
-kind: ValidatingWebhookConfiguration
-metadata:
-  annotations:
-    configsync.gke.io/webhook-configuration-update: disabled
-  name: admission-webhook.configsync.gke.io
-  labels:
-    configmanagement.gke.io/system: "true"
-    configmanagement.gke.io/arch: "csmr"
-`)
-
 	nt := nomostest.New(t, nomostesting.DriftControl)
 
+	// Test starts with Admission Webhook already installed
+	nomostest.WaitForWebhookReadiness(nt)
+
 	nt.Must(nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/hello/ns.yaml", fake.NamespaceObject("hello")))
 	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add Namespace"))
 	if err := nt.WatchForAllSyncs(); err != nil {
 		nt.T.Fatal(err)
 	}
 
-	if err := os.WriteFile(filepath.Join(nt.TmpDir, "webhook.yaml"), webhook, 0644); err != nil {
-		nt.T.Fatalf("failed to create a tmp file %v", err)
-	}
-
-	// Recreate the admission webhook
-	if _, err := nt.Shell.Kubectl("replace", "-f", filepath.Join(nt.TmpDir, "webhook.yaml")); err != nil {
-		nt.T.Fatalf("failed to replace the admission webhook %v", err)
-	}
-
-	// Verify that the webhook is disabled.
-	if _, err := nt.Shell.Kubectl("delete", "ns", "hello"); err != nil {
-		nt.T.Fatalf("failed to run `kubectl delete ns hello` %v", err)
-	}
-
-	// Remove the annotation for disabling webhook
-	if _, err := nt.Shell.Kubectl("annotate", "ValidatingWebhookConfiguration", "admission-webhook.configsync.gke.io", "configsync.gke.io/webhook-configuration-update-"); err != nil {
-		nt.T.Fatalf("failed to remove the annotation in the admission webhook %v", err)
+	//helloNS := &corev1.Namespace{}
+	//if err := nt.KubeClient.Get("hello", "", helloNS); err != nil {
+	//	nt.T.Fatal(err)
+	//}
+	//version := helloNS.ResourceVersion
+
+	nomostest.StopWebhook(nt)
+
+	tg := taskgroup.New()
+	tg.Go(func() error {
+		predicates := []testpredicates.Predicate{
+			testpredicates.StatusEquals(nt.Scheme, kstatus.CurrentStatus),
+			testpredicates.DeploymentMissingEnvVar(reconcilermanager.Reconciler, reconcilermanager.WebhookEnabled),
+		}
+		return nt.Watcher.WatchObject(kinds.Deployment(),
+			core.RootReconcilerName(configsync.RootSyncName), configsync.ControllerNamespace, predicates)
+	})
+	tg.Go(func() error {
+		return nt.Watcher.WatchObject(kinds.Namespace(), "hello", "",
+			[]testpredicates.Predicate{
+				testpredicates.MissingAnnotation(metadata.DeclaredFieldsKey),
```

Comment:
It may be worth adding a validation for the presence of the annotation before stopping the webhook, so we know the annotation is removed instead of missing initially.

Reply:
Done!